text/0093-nft-standard.md

```diff
@@ -0,0 +1,260 @@
+
+    /// Emitted when a specific NFT is approved to an account
+    Approval { owner: AccountHash, approved: Option<AccountHash>, token_id: U256 },
+
+    /// Emitted when an operator is approved/disapproved for all NFTs of an owner
+    ApprovalForAll { owner: AccountHash, operator: AccountHash, approved: bool },
+}
+
+/// Casper-compatible NFT interface
+pub trait CEP95 {
+    /// Returns the number of NFTs owned by a given account
+    ///
+    /// @param owner - The account to query.
+    /// @return U256 - The number of NFTs owned.
+    fn balance_of(&self, owner: AccountHash) -> U256;
+
+    /// Returns the owner of a specific NFT.
+    ///
+    /// @param token_id - The ID of the NFT.
+    /// @return Option<AccountHash> - The owner if it exists, else None.
+    fn owner_of(&self, token_id: U256) -> Option<AccountHash>;
+
+    /// Transfers the ownership of an NFT and performs a recipient check.
+    ///
+    /// @param from - The current owner of the NFT.
+    /// @param to - The new owner.
+    /// @param token_id - The NFT ID.
+    /// @param data - Optional payload to pass to a receiving contract.
+    fn safe_transfer_from(&mut self, from: AccountHash, to: AccountHash, token_id: U256, data: Option<Vec<u8>>);
+
+    /// Transfers the ownership of an NFT without checking the recipient contract.
+    ///
+    /// @param from - The current owner of the NFT.
+    /// @param to - The new owner.
+    /// @param token_id - The NFT ID.
+    fn transfer_from(&mut self, from: AccountHash, to: AccountHash, token_id: U256);
+
+    /// Approves another account to transfer a specific NFT.
+    ///
+    /// @param to - The account that will be granted approval.
+    /// @param token_id - The NFT ID.
+    fn approve(&mut self, to: Option<AccountHash>, token_id: U256);
+
+    /// Revokes approval for a specific NFT.
+    ///
+    /// @param token_id - The NFT ID.
+    fn revoke_approval(&mut self, token_id: U256);
+
+    /// Enables operator approval for all of the caller's NFTs.
+    ///
+    /// @param operator - The operator address to be approved.
+    fn approve_for_all(&mut self, operator: AccountHash);
+
+    /// Revokes operator approval for all of the caller's NFTs.
+    ///
+    /// @param operator - The operator address whose approval is to be revoked.
+    fn revoke_approval_for_all(&mut self, operator: AccountHash);
+
+    /// Gets the approved account for a specific NFT.
+    ///
+    /// @param token_id - The NFT ID.
+    /// @return Option<AccountHash> - Approved account if one exists.
+    fn get_approved(&self, token_id: U256) -> Option<AccountHash>;
+
+    /// Checks if an operator is approved to manage all NFTs of the owner.
+    ///
+    /// @param owner - The NFT owner's address.
+    /// @param operator - The operator to check.
+    /// @return bool - True if the operator is approved for all NFTs, false otherwise.
+    fn is_approved_for_all(&self, owner: AccountHash, operator: AccountHash) -> bool;
```

Comment:
I'd consider naming it `has_approved_for_all` like "has the owner approved operator"